app/build.gradle

```diff
@@ -1,12 +1,11 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.0"
+    compileSdkVersion 27
     defaultConfig {
         applicationId "tabian.com.instagramclone2"
         minSdkVersion 18
```

Comment:
Are you sure having minSdkVersion 21?
As I know you cannot use lambdas or other cool features using minSdkVerison 18